iceoryx_examples/request_response/README.md

```diff
@@ -0,0 +1,214 @@
+# request_response
+
+## Introduction
+
+This example demonstrates how to use iceoryx in a client-server architecture
+using the request-response communication pattern.
+
+## Expected output basic server-client example
+
+[![asciicast](https://asciinema.org/a/469913.svg)](https://asciinema.org/a/469913)
+
+## Code walkthrough
+
+In the following scenario the client (client_cxx_waitset.cpp) is using the `Waitset` to wait for a response from the server
+(server_cxx_listener.cpp) that uses the `Listener` API for taking and processing the requests.
+
+The client is inspired by the `iox-cpp-waitset-basic` example from the `waitset` folder and the server from the
+`iox-cpp-callbacks-subscriber` example in the `callbacks` folder.
+
+This is the most recommended way to create an efficient server-client combination with iceoryx.
+
+### Client using Waitset
+
+At first, the includes for the client port, request-response types, WaitSet, and runtime are needed.
+<!-- [geoffrey] [geoffrey][iceoryx_examples/request_response/client_cxx_waitset.cpp] [iceoryx includes] -->
+```cpp
+#include "request_and_response_types.hpp"
+
+#include "iceoryx_hoofs/posix_wrapper/signal_watcher.hpp"
+#include "iceoryx_posh/popo/client.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+```
+
+Next, the iceoryx runtime is initialized. With this call,
+the application will be registered at `RouDi`, the routing and discovery daemon.
+<!-- [geoffrey] [iceoryx_examples/request_response/client_cxx_waitset.cpp] [initialize runtime] -->
+```cpp
+iox::runtime::PoshRuntime::initRuntime("iox-cpp-waitset-basic");
+```
+
+After creating the runtime, the client port is created and attached to the Waitset.
+
+<!--[geoffrey][iceoryx_examples/request_response/client_cxx_waitset.cpp][create client]-->
+```cpp
+iox::popo::Client<AddRequest, AddResponse> client({"Example", "Request-Response", "Add"});
```

Comment:
Here I would hint the user that it is possible to attach custom options to the client and when they are not specified the default options are used.
Then the user is aware that the possibility of customization exists. Maybe with a hint to the header.